internal/source/source.go

```diff
@@ -169,13 +175,27 @@ func (s *Source) Open(ctx context.Context, position sdk.Position) error {
 		return fmt.Errorf("init jetstream iterator: %w", err)
 	}
 
+	// Async handlers & callbacks
+	conn.SetErrorHandler(internal.ErrorHandlerCallback(ctx))
+	conn.SetDisconnectErrHandler(internal.DisconnectErrCallback(ctx, func(c *nats.Conn) {
+		s.iterator.unAckAll()
+		s.iterator.jetstream.UpdateConsumer(s.iterator.params.Stream, &nats.ConsumerConfig{
```

Comment:
What's the intention behind this?

Reply:
https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/consumers#stream-vs-consumer-sequence-numbers

The consumer count gets zeroed when reconnects. The Stream counter still persists.